accepted/2025/dotnet-package-download.md

```diff
@@ -0,0 +1,98 @@
+* CI/CD workflows that consume `.nupkg` artifacts directly.
+* **Cross-platform builds** where many customers today rely on `nuget.exe install` in CI scripts running on Linux or macOS, forcing them to install Mono just to get this functionality.
+
+Providing `dotnet package download` improves developer productivity, simplifies pipelines, and ensures a cross-platform experience within the .NET CLI.
+
+## Explanation
+
+### Command Overview
+
+```ps1
+Description:
+  Downloads a NuGet package to a local folder without requiring a project file.
+
+Usage:
+  dotnet package download <PackageId> [options]
+
+Arguments:
+  PackageId    Package in the form of a package identifier (e.g. 'Newtonsoft.Json') 
+               or identifier and version separated by '@' (e.g. 'Newtonsoft.Json@13.0.3').
+
+Options:
+  -?, -h, --help                Show command line help.
+  --allow-insecure-connections  Allows downloading from HTTP sources.
+  --configfile <path>           Path to a NuGet.config to use.
+  --download-only               Download only the .nupkg file without extraction.
+  --interactive                 Enables interactive authentication if required.
+  --audit <mode>                Set log level for vulnerability report. Mode values: `off`, `warn`, `error`
+  --output-directory <path>     Directory where the package will be placed. Defaults to the current working directory.
+  --prerelease                  Allows downloading prerelease versions.
+  -s --source <package source>     Specifies the NuGet package source to use.
+  -v --verbosity <level>           Set the verbosity of the command. Allowed values: quiet, normal, detailed.
+```
+
+### Default Behavior
+
+By default, `dotnet package download`:
+
+* Downloads **only the explicitly requested package** (no dependencies).
+* Places the package in the **current working directory** unless `--output-directory` is specified.
+* Stores each package in its **own folder named after the package ID**, with subfolders for each version.
+
+Example:
+
+```ps1
+<output-directory>/
+  Contoso/
+    13.0.3/
+    12.0.3/
+```
+
+This layout allows multiple versions of the same package to coexist and makes it easy for scripts or tools to locate the correct version.
+
+#### Audit
+
+By default, this command runs in **audit mode set to `warn`**.
+
+* If a downloaded package is identified as vulnerable, a vulnerability message is written to **stderr**, but the exit code remains **0** (as long as the download itself was successful).
+* `--audit off`: Disables vulnerability auditing.
+* `--audit error`: Writes vulnerability messages to **stderr** and sets the exit code to **1** if any vulnerabilities are found.
+
+#### Exit Codes
+
+The command follows standard exit code conventions:
+
+* **0** – The package was successfully downloaded, or the requested version was already present.
+* **1** – The download failed, or audit mode was set to `error` and vulnerabilities were detected.
```

Comment:
System.CommandLine also uses exit code 1 when CLI arguments fail parsing.

Some other obvious errors are things like network errors, or if the package source can't find the package or the requested version. Rather than trying to enumerate every possible error, you could just say "All other errors"